db_helper/db_helper.go

```diff
@@ -0,0 +1,198 @@
+package db_helper
+
+import (
+	"errors"
+	"fmt"
+	"github.com/soltanoff/go_github_release_monitor_bot/models"
+	"github.com/soltanoff/go_github_release_monitor_bot/utils"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+	"strings"
+)
+
+func getDbConnection() *gorm.DB {
+	db, err := gorm.Open(sqlite.Open(utils.DbName), &gorm.Config{})
+	if err != nil {
+		utils.LogError("DB connection error: %s", err.Error())
+		panic("failed to connect database")
+	}
+	return db
+}
+
+func AutoMigrate() {
+	db := getDbConnection()
+	err := db.AutoMigrate(&models.User{}, &models.Repository{}, &models.UserRepository{})
+	if err != nil {
+		utils.LogError("DB migration error: %s", err.Error())
+		panic("failed to migrate database schema")
+	}
+}
+
+func GetOrCreateUser(userExternalID int64) (user *models.User, err error) {
+	db := getDbConnection()
+	tx := db.Begin()
+	defer tx.Rollback()
```

Comment:
Если ты хочешь заниматься context cancellation/gracefull shutdown, то хорошо бы прокидывать контекст первым аргументом в методы работы с БД, например: 
```suggestion
func GetOrCreateUser(ctx context.Context, userExternalID int64) (user *models.User, err error) {
	db := getDbConnection()
	tx := db.Begin().WithContext(ctx)
	defer tx.Rollback()
```

Reply:
Годнота